### PR TITLE
Add replace functions.

### PR DESCRIPTION
**Pull request by w3rs**
Added replaceElem and replaceSeq functions
```
replaceElem old new = omap (\x -> if x == old then new else x)
replaceSeq old new = ointercalate new . splitSeq old
```
and rewrite rules for Text.

### CODE REVIEW

**Comment by snoyberg** [comment]
This looks good, but can I ask for:

* An added test case. For historical reasons, the tests are mostly in classy-prelude, so if you want to add it there, that's fine, but adding in mono-traversable itself is fine too
* A .cabal version bump/changelog/`@since` comment

**Comment by w3rs** [comment]
@snoyberg I added a test case but please review it. It seems a bit too trivial for me (2 of 3 tests just check that function is same as its definition). Besides that, are rewrite rules fired for tests?

**Comment by w3rs** [comment]
also CI build fails on `lts-2` because it doesn't have `sublistOf` in QuickCheck. Not sure if we can ignore this.

File: mono-traversable/src/Data/Sequences.hs [hunk 1]
```diff
@@ -1192,6 +1192,15 @@ splitElem x = splitWhen (== x)
 splitSeq :: (IsSequence seq, Eq (Element seq)) => seq -> seq -> [seq]
 splitSeq sep = List.map fromList . List.splitOn (otoList sep) . otoList
 
+-- | @'replaceSeq' old new@ replaces all @old@ subsequences with @new@.
+--
+-- > replaceSeq old new === ointercalate new . splitSeq old
+--
+-- | @since 1.0.1
```
Comment (snoyberg) [inline mono-traversable/src/Data/Sequences.hs]: The extra `|` here isn't needed.

**Comment by snoyberg** [comment]
For `sublistOf`, how about using `#if MIN_VERSION_QuickCheck(...)` in the test suite to make it compile with old and new LTSes. That way we still sanity check old LTS versions, but get the extra tests too.

**Comment by snoyberg** [comment]
Also, I'm OK with this level of complexity for the tests, thank you for adding them 👍

**Comment by w3rs** [comment]
I amended the last commit.

Some checks still fail with

> ghc-pkg: streamingProcess: runInteractiveProcess: pipe: resource exhausted (Too many open files)

but I guess it's some kind of a system issue.

**Comment by snoyberg** [comment]
Yes, just a system issue, I'll restart the builds.

**Pull request merged**